mongo/mongo.go

```diff
@@ -106,13 +106,22 @@ func (m *Mongo) GetBooks(ctx context.Context) (models.Books, error) {
 
 	list := session.DB(m.Database).C(m.BooksCollection).Find(nil)
 
-	books := &models.Books{}
-	if err := list.All(&books.Items); err != nil {
+	totalCount, err := list.Count()
+	if err != nil {
+		log.Event(ctx, "error counting items", log.ERROR, log.Error(err))
+		if err == mgo.ErrNotFound {
+			return []*models.Book{}, totalCount, nil
+		}
+	}
+	iter := list.Sort().Skip(offset).Limit(limit).Iter()
+
+	var books []*models.Book
+	if err := iter.All(&books); err != nil {
 		log.Event(ctx, "unable to retrieve books", log.ERROR, log.Error(err), logData)
```

Comment:
Not sure the log should be an error event, likely warn as the request made by user is valid and nothing has gone wrong within the application, just that the items array cannot be populated, e.g. user has paginated passed the last book in list.

So `log.Event(ctx, "failed to iterate over books", log.WARN, log.Error(err), logData)`